pkg/iam/iam.go

```diff
@@ -0,0 +1,153 @@
+package iam
```

Comment:
There's no need to throw things in /pkg -- some folks did this pre Go 1.5 to deal with the lack of vendoring support. You can just throw things in folders. Generally you want to avoid separate pkgs unless there's a strong need for isolation between components. 

This might be a good usecase, but it will complicate things if you want to share functions between these pkgs. Go imports are always one way and cyclical imports will cause compile time failures.